### PR TITLE
Added support for hardware encoding

### PR DESCRIPTION
**Pull request by CDaut**
One can now specify an encoder ffmpeg should use, with the -e flag. Checks if the ancoder is supported and only uses if if that's the case.

### CODE REVIEW

**Comment by CDaut** [comment]
TODO: add support for hwaccel_device to select encoding device.

**Comment by ShayBox** [comment]
Thanks for making this! let me know when you're ready for it to be merged, and if you could add a section to the readme to explain how to use it.

File: src/libraries/options.ts [hunk 1]
```diff
@@ -54,7 +54,7 @@ export default class Options {
 					"-w   --wayland     Force the use of the wayland backend",
 					"-x   --x11         Force the use of the X11 backend",
 					"-v,  --verbose,    Show verbose output",
-					"-e, --hwenc,      Use hardware encoding",
+					"-e, --hwenc,      Use hardware encoding. You can get a list of supported encoders using \"ffmpec -hwaccels\"",
```
Comment (botiapa) [inline src/libraries/options.ts:57]: typo: ffmpec -> ffmpeg

Comment (CDaut) [reply src/libraries/options.ts:57]: Done.

**Pull request merged**